packages/react/src/primitives/View/View.tsx

```diff
@@ -21,15 +21,27 @@ const ViewPrimitive = <Element extends ElementType = 'div'>(
 ) => {
   const { propStyles, nonStyleProps } = useStyles(rest, style);
 
+  // No need to do extra work on style props when it is not rendered as a HTML element
+  // e.g. <Image as={NextImage} src="" width={300} height={300} />, just let all props pass to NextImage
+  const _rest =
+    typeof as === 'string'
+      ? {
+          style: propStyles,
+          ...nonStyleProps,
+        }
+      : {
+          style,
+          ...rest,
```

Comment:
Does this mean all our style props will not work with the NextImage?

Reply:
Discussed in dev sync